### DISCUSSION TITLE
Update Svc/StaticMemory to have configurable memory by instance

### DISCUSSION THREAD

**Discussion by timcanham** [nasa/fprime#1285] [Call for Comments]
`Svc/StaticMemory` currently has a configuration variable `STATIC_MEMORY_ALLOCATION_SIZE` that sets the size of the buffer. It can be done per deployment by setting that variable in `StaticMemoryConfig.hpp`, but the component would be generally more useful if each instance could be configured to have its own size. Recommendation would be to have an `initialize(Fw::MemAllocator alloctor` public function that could delegate to the user how to allocate memory, but use `Fw::MallocAllocator` in `Ref` as an example.

**Comment by ThibFrgsGmz** [comment]
This seems to me to be a good idea. It would leave more configurability to the user.
